framework/validators/CEmailValidator.php

```diff
@@ -156,15 +162,24 @@ protected function checkMxPorts($domain)
 		$records=dns_get_record($domain, DNS_MX);
 		if($records===false || empty($records))
 			return false;
+		if (is_int($this->timeout)) {
+			$timeout=$this->timeout;
+		}else {
+			$timeout=((int)ini_get("default_socket_timeout"))*1000;
+		}
 		usort($records,array($this,'mxSort'));
 		foreach($records as $record)
 		{
-			$handle=@fsockopen($record['target'],25);
-			if($handle!==false)
+			$url='smtp://'.$record['target'].':25/';
+			$ch=curl_init($url);
```

Comment:
why would you use curl to connect to an SMTP server? `fsockopen` has a timeout parameter if the issue is about adding a timeout only: http://php.net/manual/en/function.fsockopen.php